Firestore/Example/FuzzTests/FuzzingResources/Serializer/Corpus/ConvertTextToBinary.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# Copyright 2018 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Converts text protos to binary protos. Text protos are retrieved from the
+# folder defined by SCRIPT_INPUT_FILE_0 and the generated binary protos are
+# stored in the folder defined by SCRIPT_OUTPUT_FILE_0. Both SCRIPT_INPUT_FILE_0
+# and SCRIPT_OUTPUT_FILE_0 are defined in the Run Script Build Phase of the
+# XCode build target Firestore_FuzzTests_iOS that executes this script. XCode
+# defines these environment variables and makes them available to the script.
+
+# Directory that contains the text protos to convert to binary protos.
+text_protos_dir="${SCRIPT_INPUT_FILE_0}"
+
+# Create a folder to write binary protos to. This is our corpus.
+binary_protos_dir="${SCRIPT_OUTPUT_FILE_0}"
+mkdir -p "${binary_protos_dir}"
+
+echo "Converting text proto files in directory: $text_protos_dir"
+echo "Writing binary proto files to directory: $binary_protos_dir"
+
+# Run proto conversion command for each file content.
+for text_proto_file in "${text_protos_dir}"/*
+do
+  file_name="$(basename -- ${text_proto_file})"
+  file_content="$(cat ${text_proto_file})"
+
+  # Choose an appropriate message type depending on the prefix of the file.
+  message_type="Value"
+  if [[ $file_name == doc-* ]]; then
+    message_type="Document"
+  elif [[ $file_name == fv-* ]]; then
+    message_type="Value"
+  elif [[ $file_name == arr-* ]]; then
+    message_type="ArrayValue"
+  elif [[ $file_name == map-* ]]; then
+    message_type="MapValue"
+  fi
+
+  # Run the conversion.
+  echo "Converting file: ${file_name} (type: ${message_type})"
+  echo "${file_content}" \
+    | "${SRCROOT}/../../build/external/protobuf/src/protobuf-build/src/protoc" \
```

Comment:
This relies on the cmake build having been run. But your PR uses the xcode build, right? And therefore, you won't be able to rely on that.

Solutions:
1) Do this anyway. Add instructions stating that you need to run the cmake build first if you want to do fuzzing. But this feels wrong. (might be ok temporarily; see below.)
2) Run fuzzing via cmake build and not via xcode. (Probably not ideal)
3) Add a pod dependency on protoc.

Assuming you want to do go with (3), checkout Firestore/Protos/build_protos.sh and Firestore/Protos/Podfile. The latter will demonstrate how to add a pod dependency on protoc, the former will demonstrate how to use it.

Secondary issue: build/external/protobuf/src/protobuf-build/src/protoc is about to become the wrong path. It's about to become build/src/protobuf-build/src/protoc. Gil's working on a PR to switch that as we speak. Since there's no test automation around the fuzzing just yet, travis won't notice the conflict. So be prepared to issue a followup shortly to fix this once Gil's change lands.

For the purposes of this PR, I'm ok if you instead just add the following here instead:
```
TODO(minafarid): This currently relies on protoc as built by the cmake build. Switch this to using a pod dependency.
```

or something along those lines.

Reply:
That's very helpful. I will do as you suggested and work on the pod dependency in the following PR.